pkg/kudoctl/cmd/init/prereqs.go

```diff
@@ -223,3 +236,69 @@ func namespace(namespace string) *v1.Namespace {
 	}
 	return ns
 }
+
+// Validate whether the serviceAccount exists
+func validateServiceAccountExists(client corev1.ServiceAccountsGetter, opts Options) error {
+
+	serviceAccountExists := false
+
+	saList, err := client.ServiceAccounts(opts.Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, sa := range saList.Items {
+		if sa.Name == opts.ServiceAccount {
+			serviceAccountExists = true
+			break
+		}
+	}
+
+	if !serviceAccountExists {
+		return fmt.Errorf("Service Account %s does not exists - KUDO expects the serviceAccount to be present", opts.ServiceAccount)
+	}
+	return nil
+}
+
+// Validate whether the serviceAccount has cluster-admin role
+func validateClusterAdminRoleForSA(client kubernetes.Interface, opts Options) error {
+
+	saCheck := false
+	nsCheck := false
+	crCheck := false
+
+	// Check whether the serviceAccount has clusterrolebinding cluster-admin
+	crbs, err := client.RbacV1().ClusterRoleBindings().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, crb := range crbs.Items {
+		// Check whether the serviceAccount is part of the cluster-admin clusterrolebindings Subject's
+		for _, subject := range crb.Subjects {
+			if subject.Name == opts.ServiceAccount {
+				saCheck = true
+				if subject.Namespace == opts.Namespace {
```

Comment:
you could potentially drop the booleans here and just return errors from these branches

Reply:
One of the reasons I used the booleans here is there is a possibility I miss checking a valid object in the array. Say, I have two clusterrolebindings with the name crb1 and crb2 in that order like below example, if I return immediately after
a mismatch with namespace, I would miss out checking an actual CRB. I am not sure whether there is a better way to check this other than looping over all the object until I get a match or the last object. 
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: crb1
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-temp
subjects:
- kind: ServiceAccount
  name: testadmin
  namespace: kudo-test
```

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: crb2
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: testadmin
  namespace: kudo-system
```

Reply:
okay, let's leave it as is. Even if it was absolutely crucial we can always refactor later. Thank you for the explanation 👏

Reply:
@alenkacz - on second thought, if someone is using KUDO and trying to pass serviceAccount switch, I am sure he can understand the simplified error. I have attempted a simplified function, let me know whether it looks ok. I apologize this keeps dragging on.